openbas-api/src/main/java/io/openbas/rest/injector_contract/InjectorContractService.java

```diff
@@ -212,7 +213,24 @@ public ObjectNode getDynamicInjectorContractFieldsForInject(InjectorContract inj
         String key = field.get(CONTACT_ELEMENT_CONTENT_KEY).asText();
         if (!CONTACT_ELEMENT_CONTENT_KEY_NOT_DYNAMIC.contains(key)
             && field.hasNonNull(DEFAULT_VALUE_FIELD)) {
-          injectContent.set(key, field.get(DEFAULT_VALUE_FIELD));
+          JsonNode defaultValueNode = field.get(DEFAULT_VALUE_FIELD);
+          if (defaultValueNode != null
+              && !defaultValueNode.isNull()
+              && !defaultValueNode.asText().isEmpty()) {
+            JsonNode cardinalityValueNode = field.get(CONTRACT_ELEMENT_CONTENT_CARDINALITY);
+            if (cardinalityValueNode != null
+                && !cardinalityValueNode.isNull()
+                && !cardinalityValueNode.asText().isEmpty()) {
+              String cardinality = cardinalityValueNode.asText();
+              if (cardinality.equals(ContractCardinality.Multiple.name())) {
+                injectContent.set(key, defaultValueNode);
+              } else {
```

Comment:
```suggestion
              } else if(defaultValueNode.has(0)) {
```
Just to be sure ... ?